### ISSUE TITLE
SA0102 : False Positive (CSharp.CsParser : A syntax error has been discovered) 

### ISSUE THREAD

**Issue by brajm008** [DotNetAnalyzers/StyleCopAnalyzers#3066]
StyleCop returns a false positive with the new C# switch expression.
```
private static LogLevel ToNLogLogLevel(NpgsqlLogLevel level) =>
            level switch
            {
                NpgsqlLogLevel.Trace => LogLevel.Trace,
                NpgsqlLogLevel.Debug => LogLevel.Debug,
                NpgsqlLogLevel.Info => LogLevel.Info,
                NpgsqlLogLevel.Warn => LogLevel.Warn,
                NpgsqlLogLevel.Error => LogLevel.Error,
                NpgsqlLogLevel.Fatal => LogLevel.Fatal,
                _ => throw new ArgumentOutOfRangeException(nameof(level)),
            };
```

**Comment by sharwell** [comment]
SA0102 is not a diagnostic reported by this project. Is it possible you are using StyleCop Classic instead of StyleCop Analyzers?

**Comment by brajm008** [comment]
Thank you for the speedy reply. 
Only StyleCop.Analyzers is used within my project.

**Comment by sharwell** [comment]
Is it possible that the warning was copied incorrectly? There is no version of StyleCop Analyzers that ever reported SA0102 under any circumstance. It _sounds_ like StyleCop Classic is in use, but the only thing I can say with certainty is StyleCop Analyzers is not reporting this.

**Comment by vweijsters** [comment]
SA0102 is StyleCop classic or [StyleCop.MsBuild](https://github.com/StyleCop/StyleCop)'s way of saying that it encountered an error (or unknown construction in this case) during the parsing of the C# file. Closing this as external.
